client/components/pages/files/FilesTable.vue

```diff
@@ -1,23 +1,42 @@
 <script lang="ts" setup>
-import type { DataTableRowContextMenuEvent, DataTableRowDoubleClickEvent } from 'primevue/datatable'
+import type {
+  DataTablePageEvent,
+  DataTableRowContextMenuEvent,
+  DataTableRowDoubleClickEvent,
+  DataTableSortEvent,
+  DataTableSortMeta,
+} from 'primevue/datatable'
 import type { PContextMenu } from '#components'
 
 const { locale, t, d } = useI18n()
 
-const files = defineModel<FileRepresentationDTO[]>('files', { required: true })
-const selectedRow = ref<Nullable<FileRepresentationDTO>>(null)
 const selectedRows = defineModel<Array<FileRepresentationDTO>>('selectedRows', { default: () => [] })
+const selectedRow = ref<Nullable<FileRepresentationDTO>>(null)
+const page = ref(0)
+const pageSize = ref(10)
+const pageDefinition = ref<PageDefinitionDTO>({})
+
+const { data, pending, error, refresh } = useLazyFetchAPI<PageableDTO>('/files/search', {
+  method: 'POST',
+  body: pageDefinition,
+  query: { page, pageSize },
+})
+
+const files = computed(() => data.value?.page.items ?? [])
+const rows = computed(() => data.value?.page.pageSize ?? pageSize.value)
+const totalRecords = computed(() => data.value?.page.totalItems ?? 0)
 const selectedRowId = computed(() => selectedRow.value?.id)
+const errorStatus = computed(() => error.value?.statusCode)
```

Comment:
overkill honestly, you use it only once, in a place where the ref is unwrapped anyway. i think we should avoid cluttering the namespace and the symbols created